apps/namada-interface/src/hooks/useExtensionConnect.ts

```diff
@@ -0,0 +1,71 @@
+import { useIntegrationConnection } from "@namada/integrations";
+import { Account, Chain } from "@namada/types";
+import { useAtom } from "jotai";
+import { useEffect, useState } from "react";
+import { addAccounts, fetchBalances } from "slices/accounts";
+import { namadaExtensionConnectedAtom, setIsConnected } from "slices/settings";
+import { useAppDispatch } from "store";
+
+export enum ConnectStatus {
+  IDLE,
+  CONNECTING,
+  CONNECTED,
+  ERROR,
+}
+
+type UseConnectOutput = {
+  connectionStatus: ConnectStatus;
+  connect: () => Promise<void>;
+};
+
+export const useExtensionConnect = (chain: Chain): UseConnectOutput => {
```

Comment:
I would vote for having a single source of truth for the extension's connection status instead of using both `namadaExtensionConnectedAtom` and `connectionStatus`. Otherwise, the connection code in other places can get a bit complicated.

It might be that the type of `namadaExtensionConnectedAtom` should change to the `ConnectStatus` type instead of `boolean`. And then you could do `useAtomValue(namadaExtensionConnectedAtom)` in TopNavigation to get the same effect (or just returning `extensionConnected` from this hook might be fine too).

I was also thinking you probably don't need to call any of the Redux code in this hook, since we don't need to support the old interface code.

Reply:
#796